SOLIDHomework/Program.cs

```diff
@@ -16,22 +16,57 @@ public class Program
         static void Main(string[] args)
         {
             OrderService orderService = new OrderService();
-                ShoppingCart shoppingCart = new ShoppingCart("US");
-            shoppingCart.Add(new OrderItem()
+            var UnitDiscountStrategy = new UnitDiscountStrategy();
+            var SpecialDiscountStrategy = new SpecialDiscountStrategy();
+            var WeightStrategy = new WeightStrategy();
+            ShoppingCart shoppingCartUnit = new ShoppingCart("US", UnitDiscountStrategy);
+            shoppingCartUnit.Add(new OrderItem()
                 {
                     Amount = 1,
-                    SeassonEndDate =  DateTime.Now,
+                    SeasonEndDate =  DateTime.Now,
                     Code =  "Test",
                     Price =  10,
                     Type = "Unit"
                 });
```

Comment:
Generate something like ShoppingApplication with run method where you will have items created with builders and calling the checkout method, and in program you just call types registration and run method